config/config.inc.php

```diff
@@ -221,7 +221,7 @@
         $customer = new Customer();
 
         /* Change the default group */
-        if (Group::isFeatureActive()) {
+        if (Group::isFeatureActive() || $customer->id == null) {
```

Comment:
As you can see, line 221 `$customer` is brand new instantiated. With no id. So, your test is always true ; it makes no difference.
But I think we have here a line of investigation : you must find why your `$customer` instance has the wrong `id_default_group` at the very moment where the code defines whether we should display taxes or not.
You should search and debug the code responsible of this decision. Find the `if` statement responsible for tax display, debug it and trail the track till you find the bug responsible for the wrong `id_default_group`.